lib/timeline/component/item/RangeItem.js

```diff
@@ -246,8 +246,7 @@ class RangeItem extends Item {
       }
     }
 
-    // add 0.5 to compensate floating-point values rounding
-    const boxWidth = Math.max(end - start + 0.5, 1);
+    const boxWidth = Math.max(end - start, 1);
```

Comment:
@yotamberk should have a look at this. He added the 0.5 compensation.

Reply:
I fixed this originally to compensate on half pixels rounding.
Why is this removed in this PR?

Reply:
It doesn't work with item horizontal margin 0. It shouldn't be necessary with `box-sizing: border-box;`

https://github.com/visjs/vis-timeline/blob/46ea275b053e2d6bbf3821460a8fec488aa1b47c/lib/timeline/component/css/item.css#L47-L50

Reply:
the CSS is not good enough. 
Checkout https://github.com/almende/vis/issues/2092
The correct solution would be to change the code to:
` const boxWidth = Math.max(Math.ceil(end - start), 1);`

Reply:
> The correct solution would be to change the code to:
> `const boxWidth = Math.max(Math.ceil(end - start), 1);`

That solution doesn't work with item horizontal margin zero. 

The case in almende/vis#2092 is a corner case when start and end dates of very few items differs only for milliseconds while majority differs in much more time. That few items fall in `Math.max` function to be less than 1 and then width of 1px is fixed without considerate start nor end dates and then the visualization are affected (blank spaces). To resolve this a propose rounding to 3 decimals instead.